examples/cosmos/lightclient/Blockchain.qnt

```diff
@@ -0,0 +1,168 @@
+//   This is a high-level specification of Tendermint blockchain
+//   that is designed specifically for the light client.
+//   Validators have the voting power of one. If you like to model various
+//   voting powers, introduce multiple copies of the same validator
+//   (do not forget to give them unique names though)
+
+module Blockchain {
+
+    pure def Min(a: int, b: int): int = if (a < b) a else b
+
+    import typedefs.* from "./typedefs"
+
+    // a set of all nodes that can act as validators (correct and faulty)
+    const AllNodes: Set[NODE]
+    // the maximal height that can be ever reached (modelling artifact)
+    const ULTIMATE_HEIGHT: int
+    // the period within which the validators are trusted
+    const TRUSTING_PERIOD: int
+
+    // possible heights
+    pure def Heights = 1.to(ULTIMATE_HEIGHT)
+
+    // A commit is just a set of nodes who have committed the block
+    pure def Commits = powerset(AllNodes)
+
+    // The set of all block headers that can be on the blockchain.
+    // This is a simplified version of the Block data structure in the
+    // actual implementation.
+    pure def BlockHeaders = 
+        tuples(Heights, Int, Commits, powerset(AllNodes), powerset(AllNodes)).map((
+            // the block height
+            height,
+            // the block timestamp in some integer units
+            time,
+            // the nodes who have voted on the previous block, the set itself
+            // instead of a hash in the implementation, only the hashes of
+            // V and NextV are stored in a block, as V and NextV are stored
+            // in the application state
+            lastCommit,
+            // the validators of this bloc. We store the validators instead of the hash
+            VS,
+            // the validators of the next block.
+            // We store the next validators instead of the hash.
+            NextVS) => { 
+                height: height, 
+                time: time,
+                lastCommit: lastCommit,
+                VS: VS,
+                NextVS: NextVS
+            }
+        )
```

Comment:
Do we actually need this definition? I think it was used for type annotation in the old TLA+ spec.